wp-includes/class-wp-service-workers.php

```diff
@@ -491,59 +482,64 @@ protected function remove_url_scheme( $schemed_url ) {
 	 * Get validated path to file.
 	 *
 	 * @param string $url Relative path.
-	 * @param bool   $allow_content_only If to allow path from content directory only. Defaults to true.
 	 * @return null|string|WP_Error
 	 */
-	protected function get_validated_file_path( $url, $allow_content_only = true ) {
```

Comment:
Sorry, just chiming in here in case folks weren't aware of this context: the location of the service worker file that gets registered is [pretty important](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API/Using_Service_Workers#Why_is_my_service_worker_failing_to_register). It needs to be located at either at the top-level of the eventual service worker scope, or if that's not possible, then you need to configure your web server to include the `Service-Worker-Allowed` response header specifying what the scope should be.

In practice, getting folks to configure their web servers "in the wild" is a bit of a pain, so serving the service worker file from a top-level URL ends up being the easiest thing to do. So if you're going to want the service worker to control all pages on a given origin, hosting it at `/service-worker.js` (rather than `/wp-content/service-worker.js`) is important.

Reply:
The service worker itself will always be served from the top-level URL, this specific method is about where the scripts that are registered via the API are allowed from. The API combines together different scripts which will always be served from the top-level URL, however, the combined script files themselves can be in a different place. This note was about the location of the scripts which will be combined into the service worker.

Reply:
Awesome! Sorry for the noise 😄